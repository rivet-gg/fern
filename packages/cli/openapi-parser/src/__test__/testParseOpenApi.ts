import { AbsoluteFilePath, join, RelativeFilePath } from "@fern-api/fs-utils";
import { CONSOLE_LOGGER } from "@fern-api/logger";
import { serialization } from "@fern-api/openapi-ir-sdk";
import { createMockTaskContext } from "@fern-api/task-context";
import { parse, SpecImportSettings } from "../parse";

const FIXTURES_PATH = join(AbsoluteFilePath.of(__dirname), RelativeFilePath.of("fixtures"));

// eslint-disable-next-line jest/no-export
export function testParseOpenAPI(
    fixtureName: string,
    openApiFilename: string,
    asyncApiFilename?: string,
    settings?: SpecImportSettings
): void {
    // eslint-disable-next-line jest/valid-title
    describe(fixtureName, () => {
        it("parse open api", async () => {
            const absolutePathToOpenAPI = join(
                FIXTURES_PATH,
                RelativeFilePath.of(fixtureName),
                RelativeFilePath.of(openApiFilename)
            );
            const absolutePathToAsyncAPI =
                asyncApiFilename != null
                    ? join(FIXTURES_PATH, RelativeFilePath.of(fixtureName), RelativeFilePath.of(asyncApiFilename))
                    : undefined;
            const specs = [];
            if (absolutePathToOpenAPI != null) {
                specs.push({
                    absoluteFilepath: absolutePathToOpenAPI,
                    absoluteFilepathToOverrides: undefined,
                    settings
                });
            }
            if (absolutePathToAsyncAPI != null) {
                specs.push({
                    absoluteFilepath: absolutePathToAsyncAPI,
                    absoluteFilepathToOverrides: undefined,
                    settings
                });
            }

            const openApiIr = await parse({
                specs,
                taskContext: createMockTaskContext({ logger: CONSOLE_LOGGER })
            });
            const openApiIrJson = await serialization.OpenApiIntermediateRepresentation.jsonOrThrow(openApiIr, {
                skipValidation: true
            });
            expect(openApiIrJson).toMatchSnapshot();
        });
    });
}
