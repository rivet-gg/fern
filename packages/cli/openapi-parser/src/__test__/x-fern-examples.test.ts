import { testParseOpenAPI } from "./testParseOpenApi";

describe("x-fern-examples", () => {
    testParseOpenAPI("x-fern-examples", "openapi.yml");
});
