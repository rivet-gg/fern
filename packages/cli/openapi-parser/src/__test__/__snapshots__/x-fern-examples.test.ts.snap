// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`x-fern-examples x-fern-examples parse open api 1`] = `
{
  "apiVersion": null,
  "basePath": null,
  "channel": [],
  "description": null,
  "endpoints": [
    {
      "audiences": [],
      "authed": false,
      "availability": null,
      "description": null,
      "errors": {},
      "examples": [
        {
          "type": "unknown",
          "value": {
            "response": {
              "body": {
                "message": "This is a test",
              },
            },
          },
        },
        {
          "type": "unknown",
          "value": {
            "response": {
              "body": {
                "id": 1,
                "name": "John Doe",
              },
            },
          },
        },
      ],
      "generatedRequestName": "GetExampleRequest",
      "headers": [],
      "idempotent": null,
      "internal": null,
      "method": "GET",
      "operationId": null,
      "pagination": null,
      "path": "/example",
      "pathParameters": [],
      "queryParameters": [],
      "request": null,
      "requestNameOverride": null,
      "response": {
        "description": "Successful response",
        "responseProperty": null,
        "schema": {
          "type": "oneOf",
          "value": {
            "availability": null,
            "description": null,
            "generatedName": "GetExampleResponse",
            "groupName": null,
            "nameOverride": null,
            "schemas": [
              {
                "additionalProperties": false,
                "allOf": [],
                "allOfPropertyConflicts": [],
                "availability": null,
                "description": null,
                "generatedName": "NotFound",
                "groupName": null,
                "nameOverride": null,
                "properties": [
                  {
                    "audiences": [],
                    "availability": null,
                    "conflict": {},
                    "generatedName": "notFoundMessage",
                    "key": "message",
                    "nameOverride": null,
                    "schema": {
                      "availability": null,
                      "description": null,
                      "generatedName": "notFoundMessage",
                      "groupName": null,
                      "nameOverride": null,
                      "type": "optional",
                      "value": {
                        "availability": null,
                        "description": null,
                        "generatedName": "NotFoundMessage",
                        "groupName": null,
                        "nameOverride": null,
                        "schema": {
                          "default": null,
                          "format": null,
                          "maxLength": null,
                          "minLength": null,
                          "pattern": null,
                          "type": "string",
                        },
                        "type": "primitive",
                      },
                    },
                  },
                ],
                "type": "object",
              },
              {
                "availability": null,
                "description": null,
                "generatedName": "GetExampleResponseOne",
                "groupName": null,
                "nameOverride": null,
                "schema": "Schema1",
                "type": "reference",
              },
              {
                "availability": null,
                "description": null,
                "generatedName": "GetExampleResponseTwo",
                "groupName": null,
                "nameOverride": null,
                "schema": "Schema2",
                "type": "reference",
              },
              {
                "additionalProperties": false,
                "allOf": [],
                "allOfPropertyConflicts": [],
                "availability": null,
                "description": null,
                "generatedName": "GetExampleResponseMessage",
                "groupName": null,
                "nameOverride": null,
                "properties": [
                  {
                    "audiences": [],
                    "availability": null,
                    "conflict": {},
                    "generatedName": "getExampleResponseMessageMessage",
                    "key": "message",
                    "nameOverride": null,
                    "schema": {
                      "availability": null,
                      "description": null,
                      "generatedName": "getExampleResponseMessageMessage",
                      "groupName": null,
                      "nameOverride": null,
                      "type": "optional",
                      "value": {
                        "availability": null,
                        "description": null,
                        "generatedName": "GetExampleResponseMessageMessage",
                        "groupName": null,
                        "nameOverride": null,
                        "schema": {
                          "default": null,
                          "format": null,
                          "maxLength": null,
                          "minLength": null,
                          "pattern": null,
                          "type": "string",
                        },
                        "type": "primitive",
                      },
                    },
                  },
                ],
                "type": "object",
              },
            ],
            "type": "undisciminated",
          },
        },
        "type": "json",
      },
      "sdkName": null,
      "server": [],
      "summary": "Get Example",
      "tags": [],
    },
  ],
  "globalHeaders": [],
  "groups": {},
  "hasEndpointsMarkedInternal": false,
  "idempotencyHeaders": [],
  "nonRequestReferencedSchemas": [
    "Schema1",
    "Schema2",
  ],
  "schemas": {
    "Schema1": {
      "additionalProperties": false,
      "allOf": [],
      "allOfPropertyConflicts": [],
      "availability": null,
      "description": null,
      "generatedName": "Schema1",
      "groupName": null,
      "nameOverride": null,
      "properties": [
        {
          "audiences": [],
          "availability": null,
          "conflict": {},
          "generatedName": "schema1Id",
          "key": "id",
          "nameOverride": null,
          "schema": {
            "availability": null,
            "description": null,
            "generatedName": "schema1Id",
            "groupName": null,
            "nameOverride": null,
            "type": "optional",
            "value": {
              "availability": null,
              "description": null,
              "generatedName": "Schema1Id",
              "groupName": null,
              "nameOverride": null,
              "schema": {
                "default": null,
                "exclusiveMaximum": null,
                "exclusiveMinimum": null,
                "maximum": null,
                "minimum": null,
                "multipleOf": null,
                "type": "int",
              },
              "type": "primitive",
            },
          },
        },
        {
          "audiences": [],
          "availability": null,
          "conflict": {},
          "generatedName": "schema1Name",
          "key": "name",
          "nameOverride": null,
          "schema": {
            "availability": null,
            "description": null,
            "generatedName": "schema1Name",
            "groupName": null,
            "nameOverride": null,
            "type": "optional",
            "value": {
              "availability": null,
              "description": null,
              "generatedName": "Schema1Name",
              "groupName": null,
              "nameOverride": null,
              "schema": {
                "default": null,
                "format": null,
                "maxLength": null,
                "minLength": null,
                "pattern": null,
                "type": "string",
              },
              "type": "primitive",
            },
          },
        },
      ],
      "type": "object",
    },
    "Schema2": {
      "additionalProperties": false,
      "allOf": [],
      "allOfPropertyConflicts": [],
      "availability": null,
      "description": null,
      "generatedName": "Schema2",
      "groupName": null,
      "nameOverride": null,
      "properties": [
        {
          "audiences": [],
          "availability": null,
          "conflict": {},
          "generatedName": "schema2Email",
          "key": "email",
          "nameOverride": null,
          "schema": {
            "availability": null,
            "description": null,
            "generatedName": "schema2Email",
            "groupName": null,
            "nameOverride": null,
            "type": "optional",
            "value": {
              "availability": null,
              "description": null,
              "generatedName": "Schema2Email",
              "groupName": null,
              "nameOverride": null,
              "schema": {
                "default": null,
                "format": null,
                "maxLength": null,
                "minLength": null,
                "pattern": null,
                "type": "string",
              },
              "type": "primitive",
            },
          },
        },
        {
          "audiences": [],
          "availability": null,
          "conflict": {},
          "generatedName": "schema2Age",
          "key": "age",
          "nameOverride": null,
          "schema": {
            "availability": null,
            "description": null,
            "generatedName": "schema2Age",
            "groupName": null,
            "nameOverride": null,
            "type": "optional",
            "value": {
              "availability": null,
              "description": null,
              "generatedName": "Schema2Age",
              "groupName": null,
              "nameOverride": null,
              "schema": {
                "default": null,
                "exclusiveMaximum": null,
                "exclusiveMinimum": null,
                "maximum": null,
                "minimum": null,
                "multipleOf": null,
                "type": "int",
              },
              "type": "primitive",
            },
          },
        },
      ],
      "type": "object",
    },
  },
  "securitySchemes": {},
  "servers": [],
  "tags": {
    "orderedTagIds": null,
    "tagsById": {},
  },
  "title": "Sample API",
  "variables": {},
  "webhooks": [],
}
`;
